DatabaseWorkshop.md

```diff
@@ -0,0 +1,185 @@
+Add the following Class to the bottom of the file, outside the `App` classes final curly brace:
+```swift
+class Persistence {
+	static func setUp() {
+		let pool = PostgreSQLConnection.createPool(host: "localhost", port: 5432, options: [.databaseName("tododb")], poolOptions: ConnectionPoolOptions(initialCapacity: 10, maxCapacity: 50, timeout: 10000))
+		Database.default = Database(pool)
+	}
+}
+```
+This code is going to create a connection pool and set it as the default database. We now need to call the `setUp()` method inside  `App`s `postInit()` method:
+```swift
+		Persistence.setUp()
+		do {
+			try ToDo.createTableSync()
+		} catch let error {
+			print("Database already exists.")
+		}
+```
+This will create a table for us. We can now begin modifying the methods to switch saving to the `todoStore` and instead save to the database.
+
+To test everything is working as it should, we can run the Kitura server and check that a table called `ToDos` is created:
+```sql
+psql tododb
+SELECT * FROM "ToDos"
+```
+This should print the columns of a table, with no data in any of the colums. We will change the methods for handling requests now to store and retrieve data from this database.
+
+### Step 6 - Adding  @escaping
+All of the handlers now need to have `@escaping` added to their function signature, as the ORM uses an escaping method. Add `@escaping` to each method, after `completion:`. For example:
+```swift
+func storeHandler(todo: ToDo, completion: @escaping (ToDo?, RequestError?) -> Void ) { ...
+func deleteAllHandler(completion: @escaping (RequestError?) -> Void ) { ...
+```
+### Step 7 - storeHandler() updates
+We will first update storeHandler() to use the database. Remove all of the code between the methods curly braces and paste in the following:
+```swift
+		var todo = todo
+		if todo.completed == nil {
+			todo.completed = false
+		}
+		todo.id = nextId
+		todo.url = "http://localhost:8080/\(nextId)"
+		nextId += 1
+		todo.save(completion)
+```
+This code is similar to the original but uses `todo.save(completion)` which saves the `todo` object to the database and then calls the methods completion handler.
+### Step 8 - deleteOne(), deleteAll(), getOne(), getAll()
+These methods are easy to update as their logic can happen with one call using the ORM! 
+```swift
+	func deleteAllHandler(completion: @escaping (RequestError?) -> Void ) {
+		ToDo.deleteAll(completion)
+	}
+
+	func deleteOneHandler(id: Int, completion: @escaping (RequestError?) -> Void ) {
+		ToDo.delete(id: id, completion)
+	}
+	
+	func getAllHandler(completion: @escaping ([ToDo]?, RequestError?) -> Void ) {
+		ToDo.findAll(completion)
+	}
+	
+	func getOneHandler(id: Int, completion: @escaping(ToDo?, RequestError?) -> Void ) {
+		ToDo.find(id: id, completion)
+	}
+```
+Each one now uses one call to the database to accomplish it's task. The code is also easy to understand just from reading the calls happening inside the methods.
```

Comment:
it's -> its. I'm not sure what we're trying to say here. Above we said the logic can happen with one call. Are we trying to say something like ...

"The code is now simpler and easier to understand because each method uses only one call to the database to accomplish its task." ?

Reply:
✅